### PR TITLE
Upgraded to pgx v5 and bump package major version to v3

### PR DESCRIPTION
**Pull request by niallnsec**
This pull requests updates the pgx dependency to the v5 branch as v4 is out of support. I have also bumped the package version to v3 as this is a breaking change.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@pckhoi plz review

File: adapter.go [hunk 1]
```diff
@@ -134,16 +134,22 @@ func (a *Adapter) LoadPolicy(model model.Model) error {
...
+		err = rows.Scan(&pType, &v0, &v1, &v2, &v3, &v4, &v5)
+		if err != nil {
+			break
+		}
+		persist.LoadPolicyLine(policyLine(pType.String, v0.String, v1.String, v2.String, v3.String, v4.String, v5.String), model)
+	}
+	rows.Close()
+	if err == nil {
+		err = rows.Err()
+	}
```
Comment (pckhoi) [inline adapter.go]: Could have replaced with `ForEachRow`.

File: adapter.go [hunk 2]
```diff
@@ -201,19 +207,26 @@ func (a *Adapter) SavePolicy(model model.Model) error {
...
+		return err
+	}
+	_, err = tx.CopyFrom(
+		context.Background(),
+		a.tableIdentifier(),
+		[]string{"id", "p_type", "v0", "v1", "v2", "v3", "v4", "v5"},
+		pgx.CopyFromRows(rows),
+	)
+	if err != nil {
+		tx.Rollback(context.Background())
```
Comment (pckhoi) [inline adapter.go]: Should replace with `pgx.BeginFunc`.

**Review by pckhoi — CHANGES_REQUESTED** [review]
Thanks for your contribution. But please replace `QueryFunc` invocations with `ForEachRow`, and `pool.BeginFunc` with `pgx.BeginFunc`. Using shortcuts mean there are less chance of introducing bugs and it will also make it easier for me as the reviewer to spot errors. Unless you have other reasons.

**Comment by niallnsec** [comment]
Sorry, my bad. I saw that the functions used previously had been removed and didn't see the top level funcs in the pgx package to replace them, which is why I reverted to the more explicit approach.

I've changed it so that it mirrors more appropriately your existing implementation.

File: adapter.go (hunk 1 above)
Comment (niallnsec) [reply adapter.go]: Updated with latest commits

File: adapter.go (hunk 2 above)
Comment (niallnsec) [reply adapter.go]: Updated with latest commits

**Comment by pckhoi** [comment]
@hsluoyz how do you feel about bumping this adapter to v3 while still using casbin v2?

**Comment by hsluoyz** [comment]
@pckhoi it's OK. Adapter version and Casbin version are not related. See: https://github.com/casbin/redis-adapter

**Comment by pckhoi** [comment]
@niallnsec looks like the Go version is incompatible with pgx v5. Please upgrade go version in go.mod to 1.18 and up.

**Comment by niallnsec** [comment]
@pckhoi I already did update the go.mod version to 1.18

**Comment by niallnsec** [comment]
I noticed the CI was failing because I hadn't bumped the Go version from 1.14 to 1.18, so I have now done that so hopefully the tests will run

**Pull request merged**